### ISSUE TITLE
Problem when using XLNet for text classification

### ISSUE THREAD

**Issue by pinguz0** [amaiya/ktrain#56]
Hello!

I've run into an issue with the model "xlnet-base-cased" for text classification using the Transformers API. I got the code to work with "bert-base-uncased", but for XLNet I get the following error when beginning to train:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-18-cc1c1391bc89> in <module>()
----> 1 learner.fit_onecycle(3e-5, 3)

12 frames
/tensorflow-2.1.0/python3.6/tensorflow_core/python/autograph/impl/api.py in wrapper(*args, **kwargs)
    235       except Exception as e:  # pylint:disable=broad-except
    236         if hasattr(e, 'ag_error_metadata'):
--> 237           raise e.ag_error_metadata.to_exception(e)
    238         else:
    239           raise

TypeError: in converted code:

    /usr/local/lib/python3.6/dist-packages/transformers/modeling_tf_xlnet.py:945 call  *
        transformer_outputs = self.transformer(inputs, **kwargs)
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/keras/engine/base_layer.py:773 __call__
        outputs = call_fn(cast_inputs, *args, **kwargs)
    /usr/local/lib/python3.6/dist-packages/transformers/modeling_tf_xlnet.py:612 call  *
        cat_ids = tf.concat([mem_pad, token_type_ids], 0)
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/util/dispatch.py:180 wrapper
        return target(*args, **kwargs)
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/array_ops.py:1517 concat
        return gen_array_ops.concat_v2(values=values, axis=axis, name=name)
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/ops/gen_array_ops.py:1126 concat_v2
        "ConcatV2", values=values, axis=axis, name=name)
    /tensorflow-2.1.0/python3.6/tensorflow_core/python/framework/op_def_library.py:440 _apply_op_helper
        raise TypeError("%s that don't all match." % prefix)

    TypeError: Tensors in list passed to 'values' of 'ConcatV2' Op have types [int32, int64] that don't all match.
```

I tried both the 20newsgroup dataset from the tutorial notebook and another dataset, same issue with both.

**Comment by amaiya** [comment]
This is an issue with the Hugging Face transformers library, not *ktrain*.  You can read [this blog post on the Transformer interface in ktrain](https://towardsdatascience.com/text-classification-with-hugging-face-transformers-in-tensorflow-2-without-tears-ee50e4f3e7ed) for more details on the XLNet issue including a possible fix.

**Comment by pinguz0** [comment]
I see. I have already read that blog post, and I am using the latest version of Transformers which has the mentioned fix implemented. I also searched for this issue in the Transformers repo issues, but did not find this particular one.

**Comment by amaiya** [comment]
After looking into this, I think this particular issue may be a bug in *ktrain*.  I am re-opening this issue until it is investigated further and resolved.  Thanks for posting.

**Comment by amaiya** [comment]
This is fixed in v0.9.2 of *ktrain*.  Verified on `xlnet-base-cased`.  

Also, as of a couple of hours ago, Hugging Face fixed the issue with Multilingual DistilBert, so this should now work, as well.
